src/progenitor/progenitordata.cpp

```diff
@@ -116,14 +133,28 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   using parthenon::HistoryOutputVar;
   parthenon::HstVar_list hst_vars = {};
   auto Mgain = [](MeshData<Real> *md) {
-    return ReduceInGain<class fluid_prim::density>(md, 0);
+    return ReduceInGain<fluid_cons::density>(md, 1, 0);
   };
   auto Qgain = [](MeshData<Real> *md) {
-    return ReduceInGain<class internal_variables::GcovHeat>(md, 0) -
-           ReduceInGain<class internal_variables::GcovCool>(md, 0);
+    return ReduceInGain<internal_variables::GcovHeat>(md, 0, 0) -
+           ReduceInGain<internal_variables::GcovCool>(md, 0, 0);
+  };
+  for (auto rc : mdot_radii) {
+    auto Mdot = [rc, LengthCGSToCode](MeshData<Real> *md) {
+      return History::CalculateMdot(md, rc, 0);
+    };
+    hst_vars.emplace_back(HistoryOutputVar(
+        HstSum, Mdot,
+        "Mdot at r = " + std::to_string(int(rc / LengthCGSToCode / 1.e5)) + "km"));
```

Comment:
With above, you can remove the conversion here. However, I don't think you should cast to int. What if you choose 400.5 km for some reason?
```suggestion
        "Mdot at r = " + std::to_string(rc) + "km"));
```

Reply:
On second thought, it's fine to cast to an int here... probably nobody will want to look at half-km here.

Reply:
yes exactly